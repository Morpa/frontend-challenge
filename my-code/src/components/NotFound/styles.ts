import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: ${theme.spacings.xxlarge};

    & img {
      width: 100%;
      height: auto;
      object-fit: cover;
      overflow: hidden;
      border-radius: ${theme.border.radius};

      ${media.greaterThan('medium')`
         max-width: 30rem;
      `}
    }
  `}
`
export const Description = styled.h3`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xlarge};
    color: ${theme.colors.lightGrey};
    margin-top: ${theme.spacings.xsmall};
  `}
`

export const Info = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xlarge};
    color: ${theme.colors.lightGrey};
    margin-top: ${theme.spacings.xsmall};

    ${media.lessThan('medium')`
      font-size: ${theme.font.sizes.medium};
    `}
  `}
`
