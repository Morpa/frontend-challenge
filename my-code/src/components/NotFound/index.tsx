import * as S from './styles'

const NotFound = () => (
  <S.Wrapper>
    <img alt="Movie not found" src="/img/not-found.jpg" />
    <S.Description>Movie not found :(</S.Description>
    <S.Info>Try to search for another movie!</S.Info>
  </S.Wrapper>
)

export default NotFound
