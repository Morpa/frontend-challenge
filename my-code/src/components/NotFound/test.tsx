import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import NotFound from '.'

describe('<NotFound />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<NotFound />)

    expect(
      screen.getByRole('img', { name: /Movie not found/i })
    ).toHaveAttribute('src', '/img/not-found.jpg')

    expect(
      screen.getByRole('heading', { name: /Movie not found/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
