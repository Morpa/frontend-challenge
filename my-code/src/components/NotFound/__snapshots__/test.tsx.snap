// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 5.6rem;
}

.c0 img {
  width: 100%;
  height: auto;
  object-fit: cover;
  overflow: hidden;
  border-radius: 0.4rem;
}

.c1 {
  font-size: 2.0rem;
  color: #7A8C99;
  margin-top: 1.6rem;
}

.c2 {
  font-size: 2.0rem;
  color: #7A8C99;
  margin-top: 1.6rem;
}

@media (min-width:768px) {
  .c0 img {
    max-width: 30rem;
  }
}

@media (max-width:768px) {
  .c2 {
    font-size: 1.6rem;
  }
}

<div
  class="c0"
>
  <img
    alt="Movie not found"
    src="/img/not-found.jpg"
  />
  <h3
    class="c1"
  >
    Movie not found :(
  </h3>
  <p
    class="c2"
  >
    Try to search for another movie!
  </p>
</div>
`;
